lib/module.js

```diff
@@ -159,7 +161,7 @@ Module._findPath = function(request, paths, isMain) {
     if (!trailingSlash) {
       const rc = stat(basePath);
       if (rc === 0) {  // File.
-        if (!isMain) {
+        if (preserveSymlinks && !isMain) {
```

Comment:
This PR accurately reflect last weeks CTC decision to use the pre-v6 behavior as the default but to preserve the new 5950 behavior behind a flag. It intentionally leaves the 5950 changes in place but puts them behind the `--preserve-symlinks` flag.